tests/std/tests/P2502R2_generator/test.cpp

```diff
@@ -95,6 +95,11 @@ void test_one(generator<R, V, A> g0, invocable<generator<R, V, A>> auto adaptor,
     assert(i == ranges::cend(expected));
 }
 
+template <class Traits, class ValueType, class ReferenceType, class RvalueReferenceType, class R, class V, class A>
+void test_one(generator<R, V, A> g0, ranges::input_range auto&& expected) {
+    return test_one<Traits, ValueType, ReferenceType, RvalueReferenceType>(move(g0), identity{}, expected);

```

Comment:
This is passing `expected` as an lvalue instead of forwarding, intentional?

Reply:
(If this is an issue, OK to merge as-is and fix in the final `<generator>` PR into `main`)

Reply:
Yes, intentional. General ranges are typically accepted by forwarding reference and handled as lvalues. The only operations are `begin` and `end`, and it doesn't make sense to call either without calling the other, so preserving rvalue-ness is mostly a waste of time. I suppose we could theoretically accept a general range and pass it to something with specialized behavior for a specific subset of types - "I take all `input_range`s but do something special if they happen to be `vector` rvalues" - and I'd `forward` in that case.

Yeah, I suppose it makes sense to `forward` in mid-layer code like this (1) to avoid needing to change mid layers when the ends change, and (2) to avoid the "you didn't `forward` this forwarding reference!" reflex.

I'll leave this open and address it in the final merge.

Reply:
Addressed in #4953.